### DISCUSSION TITLE
How is near native performance achieved

### DISCUSSION THREAD

**Discussion by gitcomit8** [sickcodes/Docker-OSX#641] [Q&A]
As I can see the docker spins up another vm of arch Linux that runs the osx-kvm. I tried to do this same but did not achieve the same performance.

**Comment by Coool** [comment]
"Run Mac OS X in Docker with near-native performance!"[^1].
What does it mean? Is it some minor performance penalty or not compared to running on Linux using QEMU with KVM? What are benefits running on Docker?

[^1]:
- https://github.com/sickcodes/Docker-OSX/discussions/562

**Comment by sickcodes** [comment]
Graphics penalty but compute can be higher than a real mac by passing through all CPU threads.

**Reply by Coool** [reply]
@sickcodes, impossible magic! How it possible to have more threads than real hardware? 🤣 I need some real true facts (measured values etc.) to believe this nonsense. Can you provide?
